versioned_docs/version-1.2/how_to_guides/format_output.mdx

```diff
@@ -0,0 +1,304 @@
+- Use `--template GO_TEMPLATE` to compute and manipulate the output data using Go template based on the chosen data format. To avoid ambiguity, this flag can only be used along with `--format go-template`.
+
+Note that both `--format` and `--template` are marked as "Experimental" in its first iteration v1.2.0.
+
+## How to format output in ORAS commands
+
+ORAS allows to use the `--format` flag to format metadata output into structured data (e.g. JSON) and optionally use the `--template` with the [Go template](https://pkg.go.dev/text/template) to manipulate the output data. See sample usages in several commands below.
+
+### oras manifest fetch
+
+For example, when using `oras manifest fetch` with the flag `--format`, the following fields should be formatted into JSON output:
+
+- `reference`: full artifact reference by digest, e.g, `$REGISTRY/$REPO@$DIGEST`
+- `mediaType`: media type of the image manifest
+- `digest`: digest of the image manifest
+- `size`: manifest file size in bytes
+- `artifactType`: the type of an artifact when the manifest is used for an artifact
+- `content`: content object includes prettified manifest output
+  - `config`: a content descriptor describes the disposition of the targeted content
+  - `layers`:  array of objects, each object in the array MUST be a descriptor
+
+See sample use cases of formatted output for `oras manifest fetch`:
+
+- Example: use `--output <file>` and `--format` at the same time, a manifest file should be produced in the filesystem and the `mediaType` value should be outputted on the console:
+
+```console
+$ oras manifest fetch $REGISTRY/$REPO:$TAG --output sample-manifest --format go-template='{{.content.config.mediaType}}'
+application/vnd.oci.empty.v1+json
+```
+
+View the content of the generated manifest within specified `sample-manifest` file. The output should be compact JSON data:
+
+```console
+$ cat sample-manifest
+{"schemaVersion":2,"mediaType":"application/vnd.oci.image.manifest.v1+json","artifactType":"application/vnd.unknown.artifact.v1","config":{"mediaType":"application/vnd.oci.empty.v1+json","digest":"sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a","size":2,"data":"e30="},"layers":[{"mediaType":"application/vnd.oci.image.layer.v1.tar","digest":"sha256:6cb759c4296e67e35b0367f3c0f51dfdb776a0c99a45f39d0476e43d82696d65","size":14477,"annotations":{"org.opencontainers.image.title":"sbom.spdx"}},{"mediaType":"application/vnd.oci.image.layer.v1.tar","digest":"sha256:54c0e84503c8790e03afe34bfc05a5ce45c933430cfd9c5f8a99d2c89f1f1b69","size":6639,"annotations":{"org.opencontainers.image.title":"scan-test-verify-image.json"}}],"annotations":{"org.opencontainers.image.created":"2023-12-15T09:41:54Z"}}
+```
+
+> [!NOTE]
```

Comment:
something might be off here - 
![image](https://github.com/oras-project/oras-www/assets/1821104/3e7f50ae-7d19-4a51-aaaa-3c7e10b4535f)

I think the other notes also need to be fixed.